qos-service-impl/src/main/java/com/palantir/atlasdb/qos/com/palantir/atlasdb/qos/agent/QosClientConfigLoader.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.qos.com.palantir.atlasdb.qos.agent;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+import com.palantir.atlasdb.qos.config.ImmutableQosClientLimitsConfig;
+import com.palantir.atlasdb.qos.config.QosClientLimitsConfig;
+
+public class QosClientConfigLoader {
+    private final Supplier<Map<String, QosClientLimitsConfig>> qosClientLimitsConfigSupplier;
+
+    public QosClientConfigLoader(
+            Supplier<Map<String, QosClientLimitsConfig>> qosClientLimitsConfigSupplier) {
+        this.qosClientLimitsConfigSupplier = qosClientLimitsConfigSupplier;
+    }
+
+    public static QosClientConfigLoader create(
+            Supplier<Map<String, QosClientLimitsConfig>> qosClientLimitsConfigSupplier) {
+        return new QosClientConfigLoader(qosClientLimitsConfigSupplier);
+    }
+
+    public QosClientLimitsConfig getConfigForClient(String client) {
+        return qosClientLimitsConfigSupplier.get()
+                .getOrDefault(client, ImmutableQosClientLimitsConfig.builder().build());
```

Comment:
nit: this is building an object on every method call - it could just be a constant